lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -574,7 +574,7 @@ class ClassMetadataInfo implements ClassMetadata
      * READ-ONLY: The definition of the discriminator column used in JOINED and SINGLE_TABLE
      * inheritance mappings.
      *
-     * @psalm-var array<string, mixed>|null
+     * @psalm-var array{name: string, fieldName: string, type: string, length?: int, columnDefinition?: string|null}|null
```

Comment:
I looked at the `setDiscriminatorColumn` and all the usage of this property in the orm code to find the possible values.